atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -100,22 +102,25 @@ def __init__(self, workflow, remote_sources=None, destdir=IMAGE_BUILD_INFO_DIR):
         super(AddImageContentManifestPlugin, self).__init__(workflow)
         self.content_manifests_dir = os.path.join(destdir, 'content_manifests')
         self.remote_sources = remote_sources
-        self.dfp = df_parser(self.workflow.df_path, workflow=self.workflow)
         fetch_maven_results = workflow.prebuild_results.get(PLUGIN_FETCH_MAVEN_KEY) or {}
         self.pnc_artifact_ids = fetch_maven_results.get('pnc_artifact_ids') or []
 
-    @property
+    @functools.cached_property
     def icm_file_name(self):
         """Determine the name for the ICM file (name-version-release.json)."""
-        labels = Labels(self.dfp.labels)
+        # parse Dockerfile for any platform, the N-V-R labels should be equal for all platforms
+        dockerfile = self.workflow.build_dir.any_platform.dockerfile_with_parent_env(
```

Comment:
can we use just 'dockerfile' ? all we need is NVR we don't need parent injection at all, I know it doesn't hurt, but might give impression that something from parent is inherited

Reply:
Do we know that the NVR labels cannot reference parent env vars? The plugin previously used `df_parser(..., workflow=self.workflow)`, which does inject parent environment

Reply:
mmm they probably could, ok then